src/python/pants/backend/scala/util_rules/versions.py

```diff
@@ -61,6 +62,20 @@ def crossversion(self, mode: ScalaCrossVersionMode) -> str:
     def binary(self) -> str:
         return self.crossversion(ScalaCrossVersionMode.BINARY)
 
+    def __eq__(self, other: Any) -> bool:
+        return (
+            isinstance(other, ScalaVersion)
+            and other.major == self.major
+            and other.minor == self.minor
+            and other.patch == self.patch
+            and other.suffix == self.suffix
+        )
+
+    def __gt__(self, other: Any) -> bool:
+        if isinstance(other, ScalaVersion):
+            return self.major > other.major or self.minor > other.minor or self.patch > other.patch
```

Comment:
Do you have a unit test for this?

To me, it looks like this assert would (wrongfully) throw:
```python
assert ScalaVersion.parse("2.2.0") > ScalaVersion.parse("1.0.5")
```

Reply:
Great, and sorry, that was a bad example (I see the issue, but failed to translate it to the example), try this:
```python
            (ScalaVersion(1, 2, 0), ScalaVersion(2, 0, 5), False),
```

Reply:
nw, I've done my fair share of 🤦🏽 myself too 😅